### PR TITLE
QuizCategory-Quiz-ReportCard Routes Added

### PR DESCRIPTION
**Pull request by parth-1372**
**Description**:

In this Pull Request, I defined the API routes required for the Quiz application, focusing on user management, authentication functionalities, quiz operations, and result generation. The following routes have been implemented:

**Management Code:**
As requested, I refactored the code for better readability and scalability.

**Routes Added**:

**User Table:**

user

**Auth Table:**

auth

**Quiz Table:**
quiz

**Quiz-Category:**
category

**ReportCard:**
report

**Testing**:
Postman API Testing: All API routes were tested using Postman to ensure they function as expected. Below are the details of the tests conducted:
[Routes-Testing.pdf](https://github.com/user-attachments/files/17228625/Routes-Testing.pdf)

### CODE REVIEW

**Comment by shubhansu-kr** [comment]
This branch has conflicts. Please resolve them. 

We dont want any implementation we just want the requrements. So basically the description which you wrote for this pr. Add that to the documentation.md file. 

That's the only file you need to change. (This will resolve the conflict as well). I will create sub task where you can implement these logics. 

Thank you for you contribution.

**Comment by parth-1372** [comment]
Hi @shubhansu-kr  I have Successfully Resolved The conflict . And add the  documentation.md file.
Checkout My latest commit . Shoud I need To make new PullRequest?

File: src/server/Routes/Auth.js [hunk 1]
```diff
@@ -1,44 +1,61 @@
 const express = require('express');
```
Comment (shubhansu-kr) [inline src/server/Routes/Auth.js:1]: We dont need any implementation just update the document.md files with the required routes.

**Comment by shubhansu-kr** [comment]
Conflict still not resolved. Please make a new PR as a whole, this one seems to be creating issues for you

**Comment by parth-1372** [comment]
Thanks For Guidance @shubhansu-kr . I have Created PR #20 Look at that.

**Comment by shubhansu-kr** [comment]
CLosing this PR. Resolved by #20

**Pull request closed without merging**